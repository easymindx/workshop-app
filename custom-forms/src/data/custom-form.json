[
  {
    "type": "text",
    "name": "name",
    "placeholder": "John",
    "label": "Name",
    "value": "Jorge"
  },
  {
    "type": "text",
    "name": "lastname",
    "placeholder": "Doe",
    "label": "Lastname",
    "value": "Mendez"
  },
  {
    "type": "email",
    "name": "email",
    "placeholder": "jondoe@mail.com",
    "label": "Email",
    "value": ""
  }
]