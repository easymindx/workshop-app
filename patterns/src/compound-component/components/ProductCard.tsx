import useCounter from '../hooks/useCounter';
import styles from '../styles/styles.module.css';
// import noImage from '../assets/no-image.jpg';

function ProductCard(): JSX.Element {
  const value = 1;
  const { counter, increaseBy } = useCounter(value);

  return (
    <div className={styles.productCard}>
      <img src="./coffee-mug.png" alt="Coffee Mug" className={styles.productImg} />
      <span className={styles.productDescription}>Coffe Mug</span>

      <div className={styles.buttonsContainer}>
        <button className={styles.buttonMinus} onClick={() => increaseBy(-value)}>
          -
        </button>
        <div className={styles.countLabel}>{counter}</div>
        <button className={styles.buttonAdd} onClick={() => increaseBy(+value)}>
          +
        </button>
      </div>
    </div>
  );
}

export default ProductCard;
