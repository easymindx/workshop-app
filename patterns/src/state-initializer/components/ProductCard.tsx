import { createContext } from 'react';

import useCounter from '../hooks/useCounter';
import styles from '../styles/styles.module.css';

import type { ProductContextProps, ProductCardStateInit } from '../../types/product';

export const ProductContext = createContext({} as ProductContextProps);
const { Provider } = ProductContext;

function ProductCard({
  product,
  children,
  className,
  onChange,
  value,
  initialValues,
}: ProductCardStateInit): JSX.Element {
  const increaseValue = 1;
  const { counter, increaseBy } = useCounter({ onChange, product, value, initialValues });

  return (
    <Provider value={{ counter, increaseBy, product, increaseValue }}>
      <div className={`${styles.productCard} ${className}`}>{children()}</div>
    </Provider>
  );
}

export default ProductCard;
