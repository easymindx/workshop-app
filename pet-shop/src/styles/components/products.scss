.Products {
  align-items: flex-start !important;
}

.Product-show-all {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  background: #191919;
  color: #fff;
  width: 370px;
  min-height: 540px;
  padding: 3rem;

  p {
    font-size: 2rem;
    font-weight: 700;
    margin: 0;
    text-transform: uppercase;
    line-height: 1.2;
  }

  button {
    bordeR: none;
    outline: none;
    background: #fff;
    width: 60px;
    height: 60px;
    border-radius: 50%;
    cursor: pointer;
  }
}

.Product-item {
  width: 370px;
  overflow: hidden;
  margin-bottom: 2rem;

  img {
    width: 100%;
    height: 450px;
    object-fit: cover;
  }
}

.Product-item-detail {
  h3, p {
    margin: 0;
  }

  h3 {
    color: #333;
    text-transform: uppercase;
  }

  p {
    font-weight: 700;
  }
}

.Product-item-cta {
  button {
    background: transparent;
    border: none;
    outline: none;
    color: #777;
    cursor: pointer;

    &:nth-of-type(3) {
      background-color: #191919;
      color: #fff;
      width: 40px;
      height: 40px;
      border-radius: 40px;

      span {
        margin-left: 2px;
        margin-top: 4px;
      }
    }
  }
}