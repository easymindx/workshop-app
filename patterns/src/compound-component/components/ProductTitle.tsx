import { useContext } from 'react';

import { ProductContext } from './ProductCard';
import styles from '../styles/styles.module.css';

type Props = {
  className?: string;
  title?: string;
};

function ProductTitle({ title, className }: Props): JSX.Element {
  const { product } = useContext(ProductContext);
  const showTitle = title || product.title;

  return <span className={`${styles.productDescription} ${className}`}>{showTitle}</span>;
}

export default ProductTitle;
