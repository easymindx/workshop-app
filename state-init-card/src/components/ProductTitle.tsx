import { useContext } from 'react';

import { ProductContext } from './ProductCard';
import styles from '../styles/styles.module.css';

import type { ProductTitleProps } from '../types/product';

function ProductTitle({ title, className }: ProductTitleProps): JSX.Element {
  const { product } = useContext(ProductContext);
  const showTitle = title || product.title;

  return <span className={`${styles.productDescription} ${className}`}>{showTitle}</span>;
}

export default ProductTitle;
