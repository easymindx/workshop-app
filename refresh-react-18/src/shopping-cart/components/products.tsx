import { AddToCartIcon } from './icons';
import { useState } from 'react';
import { useFilters } from '../hooks/use-filters';
import { products as productsMock } from '../mocks/products.json';

export function Products() {
  const [products] = useState(productsMock);
  const { filterProducts } = useFilters();

  const filteredProducts = filterProducts(products);

  return (
    <main>
      <ul className='grid grid-cols-5 gap-5 auto-rows-max'>
        {filteredProducts.map((product) => (
          <li key={product.id} className='bg-white rounded-sm pb-4'>
            <img
              src={product.thumbnail}
              alt={product.title}
              className='w-full h-[150px] object-cover'
            />
            <div className='p-4 flex items-center justify-between'>
              <strong>{product.title}</strong>
              <span>${product.price}</span>
            </div>
            <button className='flex items-center justify-center gap-3 bg-black text-white rounded-md px-4 py-1.5 text-sm m-auto'>
              Agregar al carrito
              <AddToCartIcon />
            </button>
          </li>
        ))}
      </ul>
    </main>
  );
}
