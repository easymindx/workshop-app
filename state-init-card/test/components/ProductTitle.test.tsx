import React from 'react';
import renderer from 'react-test-renderer'
import ProductTitle from '../../src/components/ProductTitle';
import ProductCard from '../../src/components/index';
import { products } from '../data/products';


describe('Product title', () => {
  test('Display the component correctly with the custom title', () => {
    const wrapper = renderer.create(
      <ProductTitle title="Test Title" />
    )

    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  test('Show the product name in the title', () => {
    const wrapper = renderer.create(
      <ProductCard product={products[0]}>
        {
          () => (
            <ProductTitle />
          )
        }
      </ProductCard>
    )

    expect(wrapper.toJSON()).toMatchSnapshot()
  })
});
