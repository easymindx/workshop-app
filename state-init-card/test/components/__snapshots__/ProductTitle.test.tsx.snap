// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product title Display the component correctly with the custom title 1`] = `
<span
  className="productDescription undefined"
>
  Test Title
</span>
`;

exports[`Product title Show the product name in the title 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
     Coffee Mug
  </span>
</div>
`;
